moveit_core/macros/include/moveit/macros/declare_ptr.h

```diff
@@ -0,0 +1,55 @@
+/*********************************************************************
```

Comment:
I'm also not too picky here, but I do feel the separate file is somewhat more clean. If you (or another maintainer) really prefers it I'll move it to the `class_forward.h` header, but otherwise I'll keep it as is.

That is partly because is is intended as a internal public macro, and not as an implementation detail of `CLASS_FORWARD`, so good naming is somewhat important. If it was just an implementation detail of CLASS_FORWARD, I wouldn't have hesitated to put it in the same file as `CLASS_FORWARD`.